pkg/apply/taskrunner/runner.go

```diff
@@ -210,7 +214,16 @@ func (b *baseRunner) run(ctx context.Context, taskQueue chan Task,
 			currentTask.ClearTimeout()
 			if msg.Err != nil {
 				b.amendTimeoutError(msg.Err)
-				return msg.Err
+				if !o.continueOnError {
```

Comment:
Is this something that can happen? It seems like we should either handle this within the tasks themselves or always handle it here (not sure if the latter is even possible).

Reply:
This change is to allow the runner continue with next task when the previous task got an error. I think it is necessary. Assume a typical call of apply-as-a-library, it includes apply tasks and prune tasks. Even if we can handle the error and continue the process within each task, we should continue with the prune tasks even some error happens in the apply tasks.

Reply:
But I think the only way an error can end up on the `TaskChannel` is if one of the tasks put it there. So could we assume it is the responsibility of each task to put non-fatal errors on the `EventChannel` rather than the `TaskChannel`? That also means that if an error comes on the `TaskChannel`, it is considered fatal and we should exit.

Reply:
Yep, the error in `TaskChannel` is from a task. I think it makes sense to not emit error to `TaskChannel` when we want continue the apply. For the errors sent to `EventChannel`, I think you mean they are sent as part of ApplyEvent/PruneEvent/DeleteEvent.